### ISSUE TITLE
rocfft-selftest Failed on Node: length 14348907 (single): when attempting Scheme: CS_L1D_TRTR

### ISSUE THREAD

**Issue by littlewu2508** [ROCmSoftwarePlatform/rocFFT#324]
### What is the expected behavior
-  rocfft-selftest pass all tests.

### What actually happens
-  Failed on Node: length 14348907 (single): when attempting Scheme: CS_L1D_TRTRT
rocfft-selftest: /tmp/portage/sci-libs/rocFFT-4.3.0/work/rocFFT-rocm-4.3.0/library/src/plan.cpp:1620: void TreeNode::build_1D(): Assertion `false' failed.

### How to reproduce
-  Build rocFFT from source and run rocfft-sleftest

### Environment
| Hardware | description |
|-----|-----|
| GPU | Vega 20 [Radeon VII] |
| CPU | AMD Ryzen 7 5800X |

| Software | version |
|-----|-----|
| Linux | 5.13.3 |
| ROCK | Upstream Kernel |
| ROCR | v4.3.0 |
|Compiler|hipcc-4.3.0|
|Googletest|[aee0f9d9](https://github.com/google/googletest/commit/aee0f9d9b5b87796ee8a0ab26b7587ec30e8858e)  (after v1.10.0|
|Boost|1.76|
|fftw|3.3.9|

**Comment by littlewu2508** [comment]
The full build log and test log is attached in this file:
[rocFFT-4.3.0-gfx906:xnack-.build.log.gz](https://github.com/ROCmSoftwarePlatform/rocFFT/files/7036676/rocFFT-4.3.0-gfx906.xnack-.build.log.gz)

**Comment by littlewu2508** [comment]
14348907 = 3^15, meaning it's prime factor only contains 3. I tried using gdb to find the cause of this failure, and found [            function_pool::has_function(fpkey(length0 / factor, precision))](https://github.com/ROCmSoftwarePlatform/rocFFT/blob/b93c40c647d23364441fb381470ba77f59947307/library/src/plan.cpp#L743) returned all false which indicates that there is no supported kernel. 

The supported vector is:
```
std::vector of length 162, capacity 256 = {4096, 4050, 4000, 3888, 3840, 3750, 3645, 3600, 3456, 3375, 3240, 3200, 3125, 3072, 3000, 2916, 2880, 2700, 2592, 2560, 2500, 2430, 2400, 2304, 2250, 2187, 2160, 2048, 2025, 2000, 1944,
  1920, 1875, 1800, 1728, 1620, 1600, 1536, 1500, 1458, 1440, 1350, 1296, 1280, 1250, 1215, 1200, 1152, 1125, 1080, 1024, 1000, 972, 960, 900, 864, 810, 800, 768, 750, 729, 720, 675, 648, 640, 625, 600, 576, 540, 512, 500, 486, 480, 450,
  432, 405, 400, 384, 375, 360, 343, 336, 324, 320, 300, 288, 270, 256, 250, 243, 240, 225, 224, 216, 208, 200, 192, 180, 176, 169, 168, 162, 160, 150, 144, 135, 128, 125, 121, 120, 112, 108, 104, 100, 96, 90, 88, 84, 81, 80, 75, 72, 64,
  60, 56, 54, 52, 50, 49, 48, 45, 44, 42, 40, 36, 32, 30, 28, 27, 26, 25, 24, 22, 21, 20, 18, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1}
```

While 14348907 = 3^15 = 3 * 4782969 = ... = 2187 * 6561, but even function_pool::has_function(fpkey(6561, rocfft_precision_single)) returned false -- 6561 is still too latge.

**Comment by littlewu2508** [comment]
There is one line when build log: `don't gen sbcc for large size`. Is that related?

**Comment by evetsso** [comment]
This is fixed by 573544b912d62285f706e01087b8d4d32f037477, which is in the develop branch but is too new for ROCm 4.3.0.  If you're able to build a commit that's at least that new, the problem should go away.
